### PR TITLE
[Still have questions to answer] Migrate gotop from @cjbassi to @xxxserxxx

### PR DESCRIPTION
**Pull request by jmuchovej**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] **Changes incomplete, see questions below** Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] **Changes incomplete, see questions below** Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] **Changes incomplete, see questions below** Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

`gotop` is no longer maintained by @cjbassi. @xxxserxxx has picked up development.

Neither xxxserxxx nor myself have maintained a Formula within `homebrew-core`. We're seeking more information on how to go about setting up these changes.

- Is there any confirmation that needs to be made about transferring repositories?
- It seems like @cjbassi supported 3 versions of MacOS – is that something that's required to be maintained by @xxxserxxx? Is there a way to notify users about upgrading and such?

There may be other questions by @xxxserxxx, but I'm not sure. Was mostly starting the conversation.

### CODE REVIEW

File: Formula/gotop.rb [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
   bottle do
     cellar :any_skip_relocation
-    sha256 "e92acc3b1327317f9a5a6a09fe841d4cdf116ebeb3b225fa73d28b10a21d0170" => :catalina
-    sha256 "fbf79c06ef6ef84bd3f3f28fba3e49308e295f816facd0f6b7f696d6789e69a9" => :mojave
-    sha256 "58606da95a9fc75bfca7646e32e1f14e90fca4600b2bffddf5a7c14fbe3f3268" => :high_sierra
+    sha256 "" => :catalina
+    sha256 "" => :mojave
+    sha256 "" => :high_sierra
   end
```
Comment (issyl0) [inline Formula/gotop.rb]: Please don't change these. They're autogenerated by our CI when we merge the PR.

File: Formula/gotop.rb [hunk 2]
```diff
@@ -1,14 +1,14 @@
 class Gotop < Formula
   desc "Terminal based graphical activity monitor inspired by gtop and vtop"
-  homepage "https://github.com/cjbassi/gotop"
-  url "https://github.com/cjbassi/gotop/archive/3.0.0.tar.gz"
-  sha256 "d5147080bb6057f0bf0900b38438e89aa066959c845bdd4c84a9c9fe478b176f"
+  homepage "https://github.com/xxxserxxx/gotop"
+  url "https://github.com/xxxserxxx/gotop/archive/3.3.0.tar.gz"
+  sha256 ""
```
Comment (issyl0) [inline Formula/gotop.rb]: This needs a sha256 hash. Download the file at `url` and run `sha256sum` on it.

File: Formula/gotop.rb [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 
   bottle do
     cellar :any_skip_relocation
-    sha256 "e92acc3b1327317f9a5a6a09fe841d4cdf116ebeb3b225fa73d28b10a21d0170" => :catalina
-    sha256 "fbf79c06ef6ef84bd3f3f28fba3e49308e295f816facd0f6b7f696d6789e69a9" => :mojave
-    sha256 "58606da95a9fc75bfca7646e32e1f14e90fca4600b2bffddf5a7c14fbe3f3268" => :high_sierra
+    sha256 "" => :catalina
+    sha256 "" => :mojave
+    sha256 "" => :high_sierra
```
Comment (dawidd6) [inline Formula/gotop.rb]: Don't do this please. Leave the bottle block as is.

File: Formula/gotop.rb (hunk 2 above)
Comment (dawidd6) [inline Formula/gotop.rb]: Set sha256 here.

**Review by issyl0 — CHANGES_REQUESTED** [review]
Thanks for this PR, @ionlights.

> It seems like @cjbassi supported 3 versions of MacOS – is that something that's required to be maintained by @xxxserxxx?

The software has to build on all of Homebrew's supported macOS versions - currently High Sierra, Mojave and Catalina. We have automated testing to confirm that the software builds from the Homebrew formula on all of those OSes.

> Is there a way to notify users about upgrading and such?

`brew install gotop` will continue to work with the new repo location. New releases are able to be updated here with the `brew bump-formula-pr` command. What the new author does with their user base in their repo is up to them - as long as it's fairly licensed, we distribute user-contributed software. Does that answer your question?

I hope that answers most of yours and @xxxserxxx's questions - if you have more, just ask.

File: Formula/gotop.rb (hunk 1 above)
Comment (ionlights) [reply Formula/gotop.rb]: Good to know. Thanks! Does @xxxserxxx just need to version bump the URL, then?

Will the Homebrew CI then compute the sha256s for each?

File: Formula/gotop.rb (hunk 2 above)
Comment (ionlights) [reply Formula/gotop.rb]: Alright, will do – please see my reply to your prior comment as well. 🙂

File: Formula/gotop.rb (hunk 1 above)
Comment (dawidd6) [reply Formula/gotop.rb]: This whole block is handled by CI.

Comment (ionlights) [reply Formula/gotop.rb]: @dawidd6 Okay, thanks! So, if you don't mind, could you check my understanding?

These two snippets need to be updated every time `gotop` has a version bump.
https://github.com/Homebrew/homebrew-core/blob/3ba0438784e7ccfa7e2bb3e5cde31516eb99b6ab/Formula/gotop.rb#L4

https://github.com/Homebrew/homebrew-core/blob/3ba0438784e7ccfa7e2bb3e5cde31516eb99b6ab/Formula/gotop.rb#L5

This snippet is governed by whatever dependencies @xxxserxxx introduces, __and are present in `homebrew-core`.__
https://github.com/Homebrew/homebrew-core/blob/3ba0438784e7ccfa7e2bb3e5cde31516eb99b6ab/Formula/gotop.rb#L14

This snippet is based on whatever installation steps `gotop` needs to become a system command.
https://github.com/Homebrew/homebrew-core/blob/3ba0438784e7ccfa7e2bb3e5cde31516eb99b6ab/Formula/gotop.rb#L17

The ensuing block is handled by Homebrew, not @xxxserxxx. (Don't seem able to highlight the actual block, sorry.)
https://github.com/Homebrew/homebrew-core/blob/3ba0438784e7ccfa7e2bb3e5cde31516eb99b6ab/Formula/gotop.rb#L7

Additionally, docs allude to `brew bump-formula-pr` as being the method of updating this particular Formula.

**Comment by xxxserxxx** [comment]
Thanks for the quick response @issyl0 , and thanks @ionlights for taking the lead on this.

I am building the Darwin builds in dockercore/golang-cross' cross-compiling container.  I think I can change the versions to be build by changing an environment variable during the build (MACOS_DEPLOYMENT_TARGET).  I'm currently using 10.10.0, and I think I can also set it to 10.11.  What values for the three versions should I be targetting?

@ionlights , do you think you understand the source+compile part of this?  I'm still a bit lost, but I don't need to understand it if you do.

**Comment by issyl0** [comment]
If you've never made a Homebrew package before, the [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook) is a very useful read.

**Comment by ionlights** [comment]
> If you've never made a Homebrew package before, the [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook) is a very useful read.

Heh. 😅 Thanks. I'll give this a look in a few hours. 🙂

> @ionlights , do you think you understand the source+compile part of this? I'm still a bit lost, but I don't need to understand it if you do.

@xxxserxx Are you referring to these?
https://github.com/Homebrew/homebrew-core/blob/3ba0438784e7ccfa7e2bb3e5cde31516eb99b6ab/Formula/gotop.rb#L14
https://github.com/Homebrew/homebrew-core/blob/3ba0438784e7ccfa7e2bb3e5cde31516eb99b6ab/Formula/gotop.rb#L17

For L14: `depends_on`, I believe that's sourcing existing bottles (packages) that homebrew knows about?

For L17: `install`, I believe this is just a compilation command for `go`? (I'm not too familiar with `go`, so I honestly can't comment much here.)

File: Formula/gotop.rb (hunk 1 above)
Comment (dawidd6) [reply Formula/gotop.rb]: Correct. `bump-formula-pr` is the standard way to update the formula easily.

**Comment by xxxserxxx** [comment]
@ionlights Yeah, I hadn't seen that recipe.  We need to get rid of the -mod=vendor flag.  

My earlier comment about CGO_ENABLED can be ignored; Go enables it automatically when compiling natively.

**Comment by ionlights** [comment]
Will open a new PR with actual changes.

**Pull request closed without merging**